src/main/java/com/study/realworld/domain/user/application/UserJoinService.java

```diff
@@ -18,9 +20,16 @@ public UserJoinService(final UserRepository userRepository, final PasswordEncode
     }
 
     @Transactional
-    public User join(final User user) {
-        final User encodedUser = user.encode(passwordEncoder);
-        return userRepository.save(encodedUser);
+    public User join(final User requestUser) {
+        final User user = requestUser.encode(passwordEncoder);
+        validateDuplicatedEmail(user.email());
```

Comment:
```suggestion
        validateDuplicatedEmail(user.email());
        final User user = requestUser.encode(passwordEncoder);
```
제 생각이긴 한데 encoding하는 작업보다 email checking이 우선되어야 하지 않을까요?
email이 중복되었다면 굳이 encoding작업은 진행되지 않을 수도 있어 보여요

Reply:
기존에 저장된 값이 인코딩되이 저장되어 있다고 생각해서 위와 같이 작성했습니다.
로직에 문제가 있는지 한번 봐야겠군요